api/envoy/config/core/v3/substitution_format_string.proto

```diff
@@ -16,9 +17,18 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
 // [#protodoc-title: Substitution format string]
 
+// Specifies the configuration for a formatter extension.
+message Formatter {
+  string name = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
For new extensions does the name field serve any purpose? And would we ever have any other types of config? I'm wondering if typed_config can be used in the repeated below and we can just lookup the factory based on type_url? I think we do that for new extensions or maybe I keep getting this confused?

Reply:
So I mostly followed https://github.com/envoyproxy/envoy/pull/13676, which is pretty recent. It's mostly the same:

```
  repeated FatalAction fatal_actions = 28;

...
  message FatalAction {
    // Extension specific configuration for the action. It's expected to conform
    // to the ``Envoy::Server::Configuration::FatalAction`` interface.
    core.v3.TypedExtensionConfig config = 1;
  }
```

Let me check what other extension points do.

I do think we could end up with different types of configs... cc: @dio who recently reviewed another formatter that should be an extension.

Reply:
`TypedExtensionConfig` is what I was thinking of. I think you can just do repeated `TypedExtensionConfig` below and that should be fine? We should be looking up by type_url also so see other examples for that (maybe that is already what is being done). Thank you!

/wait

Reply:
Ok np, #10908 did it that way. I'll follow that.